tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3795,30 +3592,30 @@ Status FragmentMetadata::load_footer(
     return Status::Ok();
 
   Buffer buff;
-  shared_ptr<ConstBuffer> cbuff = nullptr;
-  if (f_buff == nullptr) {
+  shared_ptr<Deserializer> deserializer = nullptr;
+  if (f_tile == nullptr) {
     has_consolidated_footer_ = false;
     RETURN_NOT_OK(read_file_footer(&buff, &footer_offset_, &footer_size_));
-    cbuff = make_shared<ConstBuffer>(HERE(), &buff);
+    deserializer = make_shared<Deserializer>(HERE(), buff.data(), buff.alloced_size());
   } else {
     footer_size_ = 0;
     footer_offset_ = offset;
     has_consolidated_footer_ = true;
-    cbuff = make_shared<ConstBuffer>(HERE(), f_buff);
-    cbuff->set_offset(offset);
+    deserializer = make_shared<Deserializer>(HERE(), f_tile->data(), f_tile->size());
+    deserializer->set_offset(offset);
```

Comment:
I think we can get rid of set_offset if we construct with:
deserializer = make_shared<Deserializer>(HERE(), f_tile->data_as<uint8_t>() + offset, f_tile->size() - offset);
?

Reply:
changed, set_offset avoided